final_project/flaskr/schema.sql

```diff
@@ -54,21 +43,22 @@ database using its coordinates
 a user can vote only once per bikerack.  */
 
 CREATE TABLE votes (
-    vote_timestamp INTEGER,
-    vote_type TEXT,
+    vote_id INTEGER PRIMARY KEY AUTOINCREMENT,
+    user_id INTEGER,
+    vote_type INTEGER,
     rack_id INTEGER NOT NULL,
     FOREIGN KEY (rack_id) REFERENCES bikeracks(rack_id)
         ON UPDATE CASCADE
         ON DELETE NO ACTION
-    CHECK (vote_type in ("upvote", "downvote"))
+    CHECK (vote_type in (-1, 1))
 );
 
-/* history tables, identical to regular table,
+/* history tables, identical to regular table except without upvote_count and downvote_count,
 When a bikerack is added to bikeracks it also gets added to bikeracks_history
 that way we can link old votes back to their bikerack */
 
 CREATE TABLE bikeracks_history (
-    rack_id INTEGER PRIMARY KEY AUTOINCREMENT,
+    rack_id INTEGER PRIMARY KEY,
```

Comment:
Does this need a UNIQUE constraint? Seems like it.

Reply:
I think PRIMARY KEY already makes it unique.

Reply:
Oh, right, that makes sense.